### PR TITLE
Add drone test and release pipelines

### PR DESCRIPTION
**Pull request by wadells**
## Description

Add minimal drone validation on pull requests, and publish tags.

## Testing Done
See the PR build -- the tag is untested as of yet.  I want to get review on the whole project first.

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    commands:
+      - make build
+  - name: test
+    image: golang:1.16
+    commands:
+      - make test
+  - name: lint
+    image: golangci/golangci-lint:v1.35.2
+    commands:
+      - golangci-lint run
```
Comment (webvictim) [inline .drone.yml]: I would generally advise running lint before tests (and possibly even before build) as it gives more useful feedback earlier in the cycle.

File: .drone.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        from_secret: QUAY_PASSWORD
+    commands:
+      - apk add --no-cache make
+      - docker login -u="$USERNAME" -p="$PASSWORD" quay.io
+      - make publish
+    volumes:
+      - name: dockersock
+        path: /var/run
+
+
```
Comment (webvictim) [inline .drone.yml]:
```suggestion
```

File: .drone.yml (hunk 1 above)
Comment (wadells) [reply .drone.yml]: Moved lint before build and test in 3a06b25.

File: .drone.yml (hunk 2 above)
Comment (wadells) [reply .drone.yml]: Removed the extra line in 3a06b25.

**Review by webvictim — APPROVED** [review]
LGTM

**Pull request merged**